Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Search.hs

```diff
@@ -353,6 +354,12 @@ buildQuote ::
   m DQuote.Quote
 buildQuote searchRequest transporterId pickupTime isScheduled mbDistance mbDuration specialLocationTag customerCancellationDues nightShiftOverlapChecking fullFarePolicy = do
   let dist = fromMaybe 0 mbDistance
+      mbNightShiftCharge = case fullFarePolicy.farePolicyDetails of
```

Comment:
This should be checked inside the calculate function, instead of being passed from here.